tools/cordova/builder.js

```diff
@@ -739,3 +739,17 @@ configuration. The key may be deprecated.`);
     }
   };
 }
+
+function createSettings(options) {
+  let settings = {};
+
+  if (options.settingsFile) {
+    try {
+      settings = JSON.parse(files.readFile(options.settingsFile, 'utf8'));
+    } catch (e) {
+      throw new Error("METEOR_SETTINGS are not valid JSON.");
```

Comment:
Let's check if `e instanceof SyntaxError` here… or better yet, use the [`optimisticReadJsonOrNull`](https://github.com/janowsiany/meteor/blob/ceb90e5c5b0f1ba3bbd73ee3090b90a0b8001ec7/tools/fs/optimistic.js#L236) function to cache the parsed JSON.